test/t8_cmesh/t8_gtest_cmesh_copy.cxx

```diff
@@ -74,5 +74,10 @@ TEST_P (cmesh_copy_equality, check_equality_of_copied_cmesh_with_original)
   EXPECT_TRUE (t8_cmesh_is_equal (cmesh_original, cmesh_copy));
 }
 
+/* TODO: I am not happy about this. Can we use something else than a global variable? */
+cmesh_generator first (0);
+cmesh_generator last ((first.get_last ()));
+cmesh_generator step (0);
+
 /* Test all cmeshes over all different inputs we get through their id */
-INSTANTIATE_TEST_SUITE_P (t8_gtest_cmesh_copy, cmesh_copy_equality, AllCmeshs);
+INSTANTIATE_TEST_SUITE_P (t8_gtest_cmesh_copy, cmesh_copy_equality, ::testing::Range (::first, ::last, ::step));
```

Comment:
Instead of creating an integer range resp. iterator I suggest to directly pass a container/iterator which gives you all cmeshes one by one.

Check out the manual: http://google.github.io/googletest/reference/testing.html#INSTANTIATE_TEST_SUITE_P
There is the `ValuesIn(container)` paramter generator. The container could implement the C++ iterator interface.
Maybe just pushing to a std::vector<CmeshGenerator> does the job or implementing a container your own as you already started to do.